tools/pillowfight.cxx

```diff
@@ -434,52 +505,46 @@ class pillowfight_app : public CLI::App
     auto query_statement{ fmt::format(query_statement_, fmt::arg("bucket_name", bucket_name_)) };
 
     bool stopping{ false };
+    auto operation_generator{ operation_generator::parse(operation_ratio_string_) };
+
     while (running.test_and_set() && !stopping) {
       std::list<
         std::pair<std::chrono::system_clock::time_point,
                   std::variant<std::future<std::pair<couchbase::error, couchbase::mutation_result>>,
                                std::future<std::pair<couchbase::error, couchbase::get_result>>,
                                std::future<std::pair<couchbase::error, couchbase::query_result>>>>>
         futures;
-      for (std::size_t i = 0; i < key_value_batch_size_; ++i) {
-        auto opcode = (dist(gen) <= chance_of_get_) ? operation::get : operation::upsert;
-        if (opcode == operation::get && known_keys.empty()) {
-          opcode = operation::upsert;
-        }
-        bool should_check_known_keys{ false };
-        switch (opcode) {
-          case operation::get:
-            should_check_known_keys = hit_chance_for_get_ > dist(gen);
+
+      auto known_keys_distribution =
+        std::uniform_int_distribution<std::size_t>(0, known_keys.size() - 1);
+      for (std::size_t i = 0; i < operation_batch_size_; ++i) {
+        auto operation = operation_generator.next_operation();
+        std::string document_id = (operation != operation::cmd_insert || known_keys.empty())
```

Comment:
I think the condition should be `operation != operation::cmd_insert && !known_keys.empty()` here

Reply:
Good catch, thank you